### PR TITLE
Improve PhysicalExpr documentation

### PR DESCRIPTION
**Pull request by alamb**
## Which issue does this PR close?
Part of #7013 

## Rationale for this change

I was working on another example, and I wanted to create a `PhysicalExpr` but  could not figure out how to do so easily from the docs (I had to go poking around in the code)

Thus I wanted to provide improved documentation for myself (and I suspect others)

## What changes are included in this PR?
1. Improved documentation on `PhysicalExpr `
2. Examples of how to create a `PhysicalExpr` and how to execute it

## Are there any user-facing changes?
Docs

### CODE REVIEW

File: datafusion/physical-expr/src/physical_expr.rs [hunk 1]
```diff
@@ -34,8 +34,65 @@ use datafusion_expr::ColumnarValue;
...
+///
 /// A Physical expression knows its type, nullability and how to evaluate itself.
+///
+/// [`PhysicalPlanner`]: https://docs.rs/datafusion/latest/datafusion/physical_planner/trait.PhysicalPlanner.html
+/// [`create_physical_expr`]: crate::create_physical_expr
+/// [`Expr`]: datafusion_expr::Expr
+///
+/// # Example: Create `PhysicalExpr` from `Expr`
+/// ```
+/// # use arrow_schema::{DataType, Field, Schema};
```
Comment (alamb) [inline datafusion/physical-expr/src/physical_expr.rs:53]: This looks pretty terrible in source form, but it renders reasonably I think

Screenshot 2024-02-09 at 7 57 06 AM

**Review by comphead — APPROVED** [review]
lgtm thanks @alamb

**Review by appletreeisyellow — APPROVED** [review]
Thank you for making a better datafusion world @alamb!

**Comment by alamb** [comment]
This is mostly annoyance driven development -- when I am annoyed about an API as a user, I want to fix / make it easier to use :)

**Pull request merged**